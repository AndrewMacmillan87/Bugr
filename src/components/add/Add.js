import React from "react";

import PropTypes from "prop-types";

import "./add.css";

const Add = ({
    id,
    extraClasses,
    placeholder,
    name,
    onChange,
    revealFunc,
    onClick,
}) => {
    const addClasses = extraClasses
        ? `add-object-container ${extraClasses}`
        : "add-object-container";

    return (
        <div className={addClasses}>
            <input
                className='object-title-input'
                id={id}
                name={name}
                type='text'
                placeholder={placeholder}
                onChange={(e) => onChange(e)}
                onFocus={(e) => (e.target.placeholder = "")}
                onBlur={(e) => (e.target.placeholder = placeholder)}
            />
            <button className='add-project-btn'>
                <i
                    className='create-object fas fa-check-circle'
                    title='Create project'
                    onClick={() => {
                        onClick();
                        revealFunc("");
                    }}></i>
            </button>
            <button className='add-project-btn'>
                <i
                    className='do-not-create-object fas fa-times-circle'
                    onClick={() => revealFunc("")}></i>
            </button>
        </div>
    );
};

Add.propTypes = {
    id: PropTypes.string,
    extraClasses: PropTypes.string,
    placeholder: PropTypes.string,
    name: PropTypes.string,
    onChange: PropTypes.func,
    revealFunc: PropTypes.func,
    onClick: PropTypes.func,
};

export default Add;
