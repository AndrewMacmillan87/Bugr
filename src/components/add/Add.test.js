import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import Add from "./Add";

describe("Add component", () => {
    test("Renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<Add />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it("Renders correctly", () => {
        const tree = renderer.create(<Add />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe("Add HTML elements", () => {
        it("should render a div element", () => {
            const wrapper = mount(<Add />);
            expect(wrapper.find("div").length).toEqual(1);
        });

        it("should render an input element", () => {
            const wrapper = mount(<Add />);
            expect(wrapper.find("input").length).toEqual(1);
        });

        it("should render 2 button elements", () => {
            const wrapper = mount(<Add />);
            expect(wrapper.find("button").length).toEqual(2);
        });
    });
});
