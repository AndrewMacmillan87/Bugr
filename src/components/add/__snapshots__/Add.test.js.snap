// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add component Renders correctly 1`] = `
<div
  className="add-object-container"
>
  <input
    className="object-title-input"
    name="project-text"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="text"
  />
  <button
    className="add-project-btn"
  >
    <i
      className="create-object fas fa-check-circle"
      title="Create project"
    />
  </button>
  <button
    className="add-project-btn"
  >
    <i
      className="do-not-create-object fas fa-times-circle"
      onClick={[Function]}
    />
  </button>
</div>
`;
